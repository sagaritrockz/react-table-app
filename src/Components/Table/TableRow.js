import React from "react";
import { array } from 'prop-types'

const TableRow = ({ data, headers }) => {
    return (
        data.map((item, i) => (
            <tr key={i}>
                {headers.map((header, ind) => {
                    if (typeof item[header] == 'object') {
                        return <td key={ind}>{item[header].value}</td>
                    } else {
                        return <td key={ind}>{item[header]}</td>
                    }
                })}
            </tr>
        ))
    );    
};

TableRow.defaultProps = {
    data: [],
    headers: []
};

TableRow.propTypes = {
    data: array.isRequired,
    headers: array.isRequired
};

export default TableRow;
