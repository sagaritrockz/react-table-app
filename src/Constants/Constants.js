const API = "https://api.worldbank.org/v2/countries/";

const FILTERS = {
    region: [
        {id: "LCN", iso2code: "ZJ", value: "Latin America & Caribbean"},
        {id: "SAS", iso2code: "8S", value: "South Asia"},
        {id: "NA", iso2code: "NA", value: "Aggregates"},
        {id: "SSF", iso2code: "ZG", value: "Sub-Saharan Africa "},
        {id: "ECS", iso2code: "Z7", value: "Europe & Central Asia"},
        {id: "MEA", iso2code: "ZQ", value: "Middle East & North Africa"},
        {id: "EAS", iso2code: "Z4", value: "East Asia & Pacific"},
        {id: "NAC", iso2code: "XU", value: "North America"}
    ],
    incomeLevel: [
        {id: "HIC", iso2code: "XD", value: "High income"},
        {id: "INX", iso2code: "XY", value: "Not classified"},
        {id: "LIC", iso2code: "XM", value: "Low income"},
        {id: "LMC", iso2code: "XN", value: "Lower middle income"},
        {id: "LMY", iso2code: "XO", value: "Low & middle income"},
        {id: "MIC", iso2code: "XP", value: "Middle income"},
        {id: "UMC", iso2code: "XT", value: "Upper middle income"}
    ],
    lendingType: [
        {id: "IBD", iso2code: "XF", value: "IBRD"},
        {id: "IDB", iso2code: "XH", value: "Blend"},
        {id: "IDX", iso2code: "XI", value: "IDA"},
        {id: "LNX", iso2code: "XX", value: "Not classified"}
    ]
}

const HEADERS = ['name', 'iso2Code', 'region', 'capitalCity', 'lendingType', 'incomeLevel']

export { FILTERS, HEADERS, API };
