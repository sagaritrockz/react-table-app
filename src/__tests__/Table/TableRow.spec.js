import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TableRow from '../../Components/Table/TableRow'
import { HEADERS } from '../../Constants/Constants'

Enzyme.configure({adapter: new Adaptar()});

describe('TableRow', () => {
    it('should render correctly', () => {
        const data = [{"id":"ABW","iso2Code":"AW","name":"Aruba","region":{"id":"LCN","iso2code":"ZJ","value":"Latin America & Caribbean "},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"HIC","iso2code":"XD","value":"High income"},"lendingType":{"id":"LNX","iso2code":"XX","value":"Not classified"},"capitalCity":"Oranjestad","longitude":"-70.0167","latitude":"12.5167"},{"id":"AFG","iso2Code":"AF","name":"Afghanistan","region":{"id":"SAS","iso2code":"8S","value":"South Asia"},"adminregion":{"id":"SAS","iso2code":"8S","value":"South Asia"},"incomeLevel":{"id":"LIC","iso2code":"XM","value":"Low income"},"lendingType":{"id":"IDX","iso2code":"XI","value":"IDA"},"capitalCity":"Kabul","longitude":"69.1761","latitude":"34.5228"},{"id":"AFR","iso2Code":"A9","name":"Africa","region":{"id":"NA","iso2code":"NA","value":"Aggregates"},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"NA","iso2code":"NA","value":"Aggregates"},"lendingType":{"id":"","iso2code":"","value":"Aggregates"},"capitalCity":"","longitude":"","latitude":""},{"id":"AGO","iso2Code":"AO","name":"Angola","region":{"id":"SSF","iso2code":"ZG","value":"Sub-Saharan Africa "},"adminregion":{"id":"SSA","iso2code":"ZF","value":"Sub-Saharan Africa (excluding high income)"},"incomeLevel":{"id":"LMC","iso2code":"XN","value":"Lower middle income"},"lendingType":{"id":"IBD","iso2code":"XF","value":"IBRD"},"capitalCity":"Luanda","longitude":"13.242","latitude":"-8.81155"},{"id":"ALB","iso2Code":"AL","name":"Albania","region":{"id":"ECS","iso2code":"Z7","value":"Europe & Central Asia"},"adminregion":{"id":"ECA","iso2code":"7E","value":"Europe & Central Asia (excluding high income)"},"incomeLevel":{"id":"UMC","iso2code":"XT","value":"Upper middle income"},"lendingType":{"id":"IBD","iso2code":"XF","value":"IBRD"},"capitalCity":"Tirane","longitude":"19.8172","latitude":"41.3317"},{"id":"AND","iso2Code":"AD","name":"Andorra","region":{"id":"ECS","iso2code":"Z7","value":"Europe & Central Asia"},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"HIC","iso2code":"XD","value":"High income"},"lendingType":{"id":"LNX","iso2code":"XX","value":"Not classified"},"capitalCity":"Andorra la Vella","longitude":"1.5218","latitude":"42.5075"},{"id":"ANR","iso2Code":"L5","name":"Andean Region","region":{"id":"NA","iso2code":"NA","value":"Aggregates"},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"NA","iso2code":"NA","value":"Aggregates"},"lendingType":{"id":"","iso2code":"","value":"Aggregates"},"capitalCity":"","longitude":"","latitude":""},{"id":"ARB","iso2Code":"1A","name":"Arab World","region":{"id":"NA","iso2code":"NA","value":"Aggregates"},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"NA","iso2code":"NA","value":"Aggregates"},"lendingType":{"id":"","iso2code":"","value":"Aggregates"},"capitalCity":"","longitude":"","latitude":""},{"id":"ARE","iso2Code":"AE","name":"United Arab Emirates","region":{"id":"MEA","iso2code":"ZQ","value":"Middle East & North Africa"},"adminregion":{"id":"","iso2code":"","value":""},"incomeLevel":{"id":"HIC","iso2code":"XD","value":"High income"},"lendingType":{"id":"LNX","iso2code":"XX","value":"Not classified"},"capitalCity":"Abu Dhabi","longitude":"54.3705","latitude":"24.4764"},{"id":"ARG","iso2Code":"AR","name":"Argentina","region":{"id":"LCN","iso2code":"ZJ","value":"Latin America & Caribbean "},"adminregion":{"id":"LAC","iso2code":"XJ","value":"Latin America & Caribbean (excluding high income)"},"incomeLevel":{"id":"HIC","iso2code":"XD","value":"High income"},"lendingType":{"id":"IBD","iso2code":"XF","value":"IBRD"},"capitalCity":"Buenos Aires","longitude":"-58.4173","latitude":"-34.6118"}];
        
        const output = shallow(
            <TableRow key={0} data={data} headers={HEADERS} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});