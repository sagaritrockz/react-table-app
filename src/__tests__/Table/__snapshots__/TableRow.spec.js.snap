// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render correctly 1`] = `
Array [
  <tr
    key="0"
>
    <td
        key="0"
    >
        Aruba
    </td>
    <td
        key="1"
    >
        AW
    </td>
    <td
        key="2"
    >
        Latin America & Caribbean 
    </td>
    <td
        key="3"
    >
        Oranjestad
    </td>
    <td
        key="4"
    >
        Not classified
    </td>
    <td
        key="5"
    >
        High income
    </td>
</tr>,
  <tr
    key="1"
>
    <td
        key="0"
    >
        Afghanistan
    </td>
    <td
        key="1"
    >
        AF
    </td>
    <td
        key="2"
    >
        South Asia
    </td>
    <td
        key="3"
    >
        Kabul
    </td>
    <td
        key="4"
    >
        IDA
    </td>
    <td
        key="5"
    >
        Low income
    </td>
</tr>,
  <tr
    key="2"
>
    <td
        key="0"
    >
        Africa
    </td>
    <td
        key="1"
    >
        A9
    </td>
    <td
        key="2"
    >
        Aggregates
    </td>
    <td
        key="3"
    />
    <td
        key="4"
    >
        Aggregates
    </td>
    <td
        key="5"
    >
        Aggregates
    </td>
</tr>,
  <tr
    key="3"
>
    <td
        key="0"
    >
        Angola
    </td>
    <td
        key="1"
    >
        AO
    </td>
    <td
        key="2"
    >
        Sub-Saharan Africa 
    </td>
    <td
        key="3"
    >
        Luanda
    </td>
    <td
        key="4"
    >
        IBRD
    </td>
    <td
        key="5"
    >
        Lower middle income
    </td>
</tr>,
  <tr
    key="4"
>
    <td
        key="0"
    >
        Albania
    </td>
    <td
        key="1"
    >
        AL
    </td>
    <td
        key="2"
    >
        Europe & Central Asia
    </td>
    <td
        key="3"
    >
        Tirane
    </td>
    <td
        key="4"
    >
        IBRD
    </td>
    <td
        key="5"
    >
        Upper middle income
    </td>
</tr>,
  <tr
    key="5"
>
    <td
        key="0"
    >
        Andorra
    </td>
    <td
        key="1"
    >
        AD
    </td>
    <td
        key="2"
    >
        Europe & Central Asia
    </td>
    <td
        key="3"
    >
        Andorra la Vella
    </td>
    <td
        key="4"
    >
        Not classified
    </td>
    <td
        key="5"
    >
        High income
    </td>
</tr>,
  <tr
    key="6"
>
    <td
        key="0"
    >
        Andean Region
    </td>
    <td
        key="1"
    >
        L5
    </td>
    <td
        key="2"
    >
        Aggregates
    </td>
    <td
        key="3"
    />
    <td
        key="4"
    >
        Aggregates
    </td>
    <td
        key="5"
    >
        Aggregates
    </td>
</tr>,
  <tr
    key="7"
>
    <td
        key="0"
    >
        Arab World
    </td>
    <td
        key="1"
    >
        1A
    </td>
    <td
        key="2"
    >
        Aggregates
    </td>
    <td
        key="3"
    />
    <td
        key="4"
    >
        Aggregates
    </td>
    <td
        key="5"
    >
        Aggregates
    </td>
</tr>,
  <tr
    key="8"
>
    <td
        key="0"
    >
        United Arab Emirates
    </td>
    <td
        key="1"
    >
        AE
    </td>
    <td
        key="2"
    >
        Middle East & North Africa
    </td>
    <td
        key="3"
    >
        Abu Dhabi
    </td>
    <td
        key="4"
    >
        Not classified
    </td>
    <td
        key="5"
    >
        High income
    </td>
</tr>,
  <tr
    key="9"
>
    <td
        key="0"
    >
        Argentina
    </td>
    <td
        key="1"
    >
        AR
    </td>
    <td
        key="2"
    >
        Latin America & Caribbean 
    </td>
    <td
        key="3"
    >
        Buenos Aires
    </td>
    <td
        key="4"
    >
        IBRD
    </td>
    <td
        key="5"
    >
        High income
    </td>
</tr>,
]
`;
